ptracer/context_helper_linux_test.go

```diff
@@ -0,0 +1,263 @@
+		t.Fatalf("Failed to start test process: %v", err)
+	}
+
+	cleanup := func() {
+		cmd.Process.Kill()
+		cmd.Wait()
+	}
+
+	return cmd.Process.Pid, cleanup
+}
+
+// TestVmRead tests the vmRead function
+func TestVmRead(t *testing.T) {
+	pid, cleanup := createTestProcess(t)
+	defer cleanup()
+
+	// create test data
+	testData := []byte("Hello, World!")
+	buff := make([]byte, len(testData))
+
+	// get process memory maps
+	maps, err := os.ReadFile(fmt.Sprintf("/proc/%d/maps", pid))
+	if err != nil {
+		t.Fatalf("Failed to read process maps: %v", err)
+	}
+
+	// parse memory maps to find a readable address
+	var addr uintptr
+	for _, line := range bytes.Split(maps, []byte{'\n'}) {
+		if len(line) == 0 {
+			continue
+		}
+		if bytes.Contains(line, []byte("r-x")) { // find a readable segment
+			var start uint64
+			fmt.Sscanf(string(line), "%x-", &start)
+			addr = uintptr(start)
+			break
+		}
+	}
+
+	if addr == 0 {
+		t.Fatal("Failed to find readable memory region")
+	}
+
+	// test reading
+	n, err := vmRead(pid, addr, buff)
+	if err != nil {
+		t.Fatalf("vmRead failed: %v", err)
+	}
+	if n == 0 {
+		t.Error("vmRead returned 0 bytes")
+	}
+}
+
+// TestVmReadStr tests the vmReadStr function
+func TestVmReadStr(t *testing.T) {
+	pid, cleanup := createTestProcess(t)
+	defer cleanup()
+
+	// test cases
+	testCases := []struct {
+		name      string
+		buffSize  int
+		addrAlign uintptr // address alignment, used to test different alignment scenarios
+		wantErr   bool
+	}{
+		{
+			name:      "small_buffer_aligned",
+			buffSize:  10,
+			addrAlign: 0,
+			wantErr:   false,
+		},
+		{
+			name:      "small_buffer_unaligned",
+			buffSize:  10,
+			addrAlign: 1,
+			wantErr:   false,
+		},
+		{
+			name:      "exact_page_size",
+			buffSize:  pageSize,
+			addrAlign: 0,
+			wantErr:   false,
+		},
+		{
+			name:      "cross_page_boundary",
+			buffSize:  pageSize + 100,
+			addrAlign: uintptr(pageSize - 50),
+			wantErr:   false,
+		},
+		{
+			name:      "large_buffer_unaligned",
+			buffSize:  pageSize * 2,
+			addrAlign: 123,
+			wantErr:   false,
+		},
+		{
+			name:      "buffer_smaller_than_to_boundary",
+			buffSize:  10,
+			addrAlign: uintptr(pageSize - 100), // distance to page boundary is 100 bytes, but buffer is only 10 bytes
+			wantErr:   false,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			buff := make([]byte, tc.buffSize)
+			
+			// get a readable memory address
+			maps, err := os.ReadFile(fmt.Sprintf("/proc/%d/maps", pid))
+			if err != nil {
+				t.Fatalf("Failed to read process maps: %v", err)
+			}
+
+			var baseAddr uintptr
+			for _, line := range bytes.Split(maps, []byte{'\n'}) {
+				if len(line) == 0 {
+					continue
+				}
+				if bytes.Contains(line, []byte("r-x")) {
+					var start uint64
+					fmt.Sscanf(string(line), "%x-", &start)
+					baseAddr = uintptr(start)
+					break
+				}
+			}
+
+			if baseAddr == 0 {
+				t.Fatal("Failed to find readable memory region")
+			}
```

Comment:
It seems like this logic is being copied multiple times, maybe consider make it a helper function.